### PR TITLE
EPMRPP-35552 React. Common logic. Log level slider

### PR DESCRIPTION
**Pull request by yana95**

### CODE REVIEW

File: app/src/components/inputs/inputSlider/inputSlider.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  onChange = (newValue) => {
+    this.setState({ value: newValue });
+    this.props.onChange(this.state.marks[newValue]);
+  };
+
+  getMaxValue = () => (this.props.options.length ? this.props.options.length - 1 : 0);
+
+  getSliderValue = () => {
```
Comment (DiscoElevator) [inline app/src/components/inputs/inputSlider/inputSlider.jsx]:
```
getSliderValue  = () => {
  const index =  this.props.options.indexOf(this.props.value);
  return index > -1 ? index : 0;
}
```

File: app/src/components/inputs/inputSlider/inputSlider.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  };
+  constructor(props) {
+    super(props);
+    this.state = {
+      marks: this.createMarks(),
+      value: this.getSliderValue() || 0,
+    };
+  }
+
+  onChange = (newValue) => {
```
Comment (DiscoElevator) [inline app/src/components/inputs/inputSlider/inputSlider.jsx]:
```
onChange => (newIndex) => this.props.onChange(this.props.options[newIndex]);
```

File: app/src/components/inputs/inputSlider/inputSlider.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    let value;
+    this.props.options.forEach((item, i) => {
+      if (item === this.props.value) {
+        value = i;
+      }
+    });
+    return value;
+  };
+
+  createMarks = () => {
```
Comment (DiscoElevator) [inline app/src/components/inputs/inputSlider/inputSlider.jsx]:
```
createMarks = () => this.prosp.options.reduce((acc, item, index) => ({ ...acc, [index]: item }), {});
```

File: app/src/components/inputs/inputSlider/inputSlider.jsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  render() {
+    return (
+      <div className={cx('slider-input')}>
+        <Slider
+          min={0}
+          max={+this.getMaxValue()}
+          marks={this.state.marks}
+          step={1}
+          value={this.state.value}
```
Comment (DiscoElevator) [inline app/src/components/inputs/inputSlider/inputSlider.jsx]: `this.getSliderValue()`

File: app/src/components/inputs/inputSlider/inputSlider.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    onChange: PropTypes.func,
+  };
+  static defaultProps = {
+    options: [],
+    value: '',
+    onChange: () => {},
+  };
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (DiscoElevator) [inline app/src/components/inputs/inputSlider/inputSlider.jsx]: There's no need to store anything in state in this case.
Caching options converted to object might be a good idea, but in this case there should be a way to sync this with props which will make this component more complicated. I don't think it's necessary at this moment.

File: app/src/components/inputs/inputSlider/inputSlider.jsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import 'rc-tooltip/assets/bootstrap.css';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './inputSlider.scss';
+
+const cx = classNames.bind(styles);
+
+export class InputSlider extends Component {
+  static propTypes = {
+    options: PropTypes.array,
```
Comment (DiscoElevator) [inline app/src/components/inputs/inputSlider/inputSlider.jsx]: PropTypes.arrayOf(PropTypes.string)

File: app/src/components/inputs/inputSlider/README.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+## **InputSlider component**
+
+Component use rc-slider library. https://github.com/react-component/slider
```
Comment (DiscoElevator) [inline app/src/components/inputs/inputSlider/README.md]: use -> uses

File: app/src/components/inputs/inputSlider/inputSlider.stories.jsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      backdrop: 'rgba(70, 69, 71, 0.2)',
+      background: '#ffffff',
+      height: 32,
+      width: 300,
+    }),
+  )
+  .addDecorator(withReadme(README))
+  .add('Default state', () => <InputSlider />)
+  .add('With options', () => <InputSlider options={options} />)
+  .add('With options, value', () => <InputSlider options={options} value={'Info'} />)
```
Comment (DiscoElevator) [inline app/src/components/inputs/inputSlider/inputSlider.stories.jsx]: value="Info"

File: app/src/components/inputs/inputSlider/inputSlider.jsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    const marks = {};
+    this.props.options.forEach((item, idx) => {
+      marks[idx] = item;
+    });
+    return marks;
+  };
+
+  render() {
+    return (
+      <div className={cx('slider-input')}>
```
Comment (ArtsiomYeliseyenka) [inline app/src/components/inputs/inputSlider/inputSlider.jsx]: `cx('input-slider')`

**Pull request merged**